### ISSUE TITLE
Document git branch/merge process

### ISSUE THREAD

**Issue by mannykary** [FightPandemics/FightPandemics#1130]
Create a draft document in Github

Related: #1127

**Comment by mannykary** [comment]
Preliminary thoughts from Slack. Not yet finalized.

I envision that we will be getting rid of the master branch and instead have a production branch which would reflect what is deployed to the production environment, and a staging branch which would reflect what is deployed to the staging environment. The feature/* branches deploy to dev/qa environments.

I think after a PR is approved for a feature/* branch, we can have QA test the build, and if it passes QA then we can merge to the staging branch. So it might make sense to add a QA approval requirement to each feature/* PR to ensure that we aren’t merging untested code into staging . And then prior to each release to production, QA would do testing in staging to make sure there are no regressions. Once regression testing passes, then we can merge staging to production and deploy to production.

**Comment by mannykary** [comment]
A thought: we could make the `staging` branch stable - so we don't merge in `feature/*` branch code directly into `staging` but rather into a `rc/v*` branch. The `rc/v*` branch will have its own review build (we'd need to add a branch matcher for this). Then when we are about to do a release to production, we first merge the `rc/v*` branch to `staging` to deploy to staging, and regression test in staging. Once regression tests pass, we merge/deploy the `staging` branch to `production` to deploy to production.

This would allow us to make critical hotfixes by branching off of `staging` (since it would mirror the code in `production` except when we are about to do a launch). So if we have a critical hotfix we need to deploy to production, we would branch off of `staging` to create a `hotfix/*` branch. Then we merge the `hotfix/*` branch into `staging` and then `production`.

Overall branch/merge process:
```

            feature/*              
                ↓               
feature/*  ➞  rc/v*  ➞  staging  ➞  production
                ↑          ↑
            feature/*   hotfix/*
```

**Comment by mannykary** [comment]
Another option is to follow the standard [git flow](https://nvie.com/posts/a-successful-git-branching-model/) model, which is quite well known. The `release/*` branch would be created to freeze the code to be released to production. This would also deploy to the staging environment. Any bug fixes would be created in a `bugfix/*` branch and merged into the `release/*` branch. After the `release/*` branch is stable and passes QA, we merge it into both `master` and `develop`.
Any critical hotfixes would need to be branched off `master`, and then merged back into `master`, `release/*` (if there is one open), and `develop` (if the branches needs the fix right away).

```
            feature/*     bugfix/*            
                ↓            ↓    
feature/*  ➞  develop  ➞  release/*  ➞  master
                ↑            ↑              ↑
                 ----------------------- hotfix/*
```

**Comment by joshmorel** [comment]
This was resolved w #1269
